[nest] NullPointerException - Could not get data model

Reported [here](https://community.openhab.org/t/nestbinding-error-172/14030):

I've recently setup a new ObenHAB server on a Raspberry Pi - all is working well apart from the binding for my Nest devices.

I've turned on logging and I can see the request is sent off, I then receive a reply that appears to be all the devices and data as I would expect to see it (names and values tie up with real world). However, after that I get a warning.... I'm assuming that the binding is able to deal with something in the data that has been returned by the nest API.

Can anyone assist with how I would debug/resolve this?

For information... I've two separate Nest home/locations setup, one has two thermostats and the other has one thermostat and two smoke detectors, all devices have unique names. At the moment I'm just trying to setup a simple read of one thermostat however I'm not getting that far as the data doesn't seem to be parsed correctly when returned from Nest.

openHAB version is 1.8.3 (latest main release)
The nest binding is 1.8.3 (latest version)
OS is Raspbian Jessie 4.4 (current version)

```
2016-09-11 21:16:44.786 TRACE o.o.b.n.i.m.AbstractRequest[:131]- About to execute 'https://developer-api.nest. ...Blah Blah

2016-09-11 21:16:46.188 TRACE o.o.b.n.i.m.AbstractRequest[:172]- {"devices":{"thermostats":{"112 ...Blah Blah

2016-09-11 21:16:46.259 WARN o.o.b.n.internal.NestBinding[:172]- Exception reading from Nest.
org.openhab.binding.nest.internal.NestException: Could not get data model.
at org.openhab.binding.nest.internal.messages.AbstractRequest.newException(AbstractRequest.java:91)
at org.openhab.binding.nest.internal.messages.DataModelRequest.execute(DataModelRequest.java:66)
at org.openhab.binding.nest.internal.NestBinding.readNest(NestBinding.java:189)
at org.openhab.binding.nest.internal.NestBinding.execute(NestBinding.java:168)
at org.openhab.core.binding.AbstractActiveBinding$BindingActiveService.execute(AbstractActiveBinding.java:156)
at org.openhab.core.service.AbstractActiveService$RefreshThread.run(AbstractActiveService.java:173)
Caused by: java.lang.NullPointerException: null
at org.openhab.binding.nest.internal.messages.Structure.sync(Structure.java:304)
at org.openhab.binding.nest.internal.messages.DataModel.sync(DataModel.java:478)
at org.openhab.binding.nest.internal.messages.DataModelRequest.execute(DataModelRequest.java:62)
at org.openhab.binding.nest.internal.NestBinding.readNest(NestBinding.java:189)
at org.openhab.binding.nest.internal.NestBinding.execute(NestBinding.java:168)
at org.openhab.core.binding.AbstractActiveBinding$BindingActiveService.execute(AbstractActiveBinding.java:156)
```

--- a/bundles/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/messages/Structure.java
+++ b/bundles/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/messages/Structure.java
@@ -31,7 +31,7 @@ public class Structure extends AbstractMessagePart implements DataModelElement {
 
     /**
      * Describes the Structure state; see the Away Guide for more information.
-     * 
+     *
      * @see <a href="https://developer.nest.com/documentation/cloud/structure-guide">Structure state</a>
      * @see <a href="https://developer.nest.com/documentation/cloud/away-guide">Away Guide</a>
      * @see <a href="https://developer.nest.com/documentation/cloud/api-overview#away">API Overview</a>
@@ -287,7 +287,7 @@ public class Structure extends AbstractMessagePart implements DataModelElement {
     }
 
     /**
-     * This method creates maps to device objects, using the list of device IDs that were unmarshalled from JSON.
+     * This method creates maps to device objects, using the list of device IDs that were deserialized from JSON.
      */
     @Override
     public void sync(DataModel dataModel) {
@@ -295,27 +295,33 @@ public class Structure extends AbstractMessagePart implements DataModelElement {
         this.thermostats_by_name = new HashMap<String, Thermostat>();
         if (this.thermostat_id_list != null) {
             for (String id : this.thermostat_id_list) {
-                Thermostat th = dataModel.getDevices().getThermostats_by_id().get(id);
-                if (th != null) {
-                    this.thermostats_by_name.put(th.getName(), th);
+                if (dataModel.getDevices() != null && dataModel.getDevices().getThermostats_by_id() != null) {
+                    Thermostat th = dataModel.getDevices().getThermostats_by_id().get(id);
+                    if (th != null) {
+                        this.thermostats_by_name.put(th.getName(), th);
+                    }
                 }
             }
         }
         this.smoke_co_alarms_by_name = new HashMap<String, SmokeCOAlarm>();
         if (this.smoke_co_alarm_id_list != null) {
             for (String id : this.smoke_co_alarm_id_list) {
-                SmokeCOAlarm sm = dataModel.getDevices().getSmoke_co_alarms_by_id().get(id);
-                if (sm != null) {
-                    this.smoke_co_alarms_by_name.put(sm.getName(), sm);
+                if (dataModel.getDevices() != null && dataModel.getDevices().getSmoke_co_alarms_by_id() != null) {
+                    SmokeCOAlarm sm = dataModel.getDevices().getSmoke_co_alarms_by_id().get(id);
+                    if (sm != null) {
+                        this.smoke_co_alarms_by_name.put(sm.getName(), sm);
+                    }
                 }
             }
         }
         this.cameras_by_name = new HashMap<String, Camera>();
         if (this.camera_id_list != null) {
             for (String id : this.camera_id_list) {
-                Camera cam = dataModel.getDevices().getCameras_by_id().get(id);
-                if (cam != null) {
-                    this.cameras_by_name.put(cam.getName(), cam);
+                if (dataModel.getDevices() != null && dataModel.getDevices().getCameras_by_id() != null) {
+                    Camera cam = dataModel.getDevices().getCameras_by_id().get(id);
+                    if (cam != null) {
+                        this.cameras_by_name.put(cam.getName(), cam);
+                    }
                 }
             }
         }

